### PR TITLE
feat: Add sum_within_month to subscription calculation

### PR DESCRIPTION
**Pull request by johevemi**

### CODE REVIEW

File: source/databricks/calculation_engine/package/calculation/wholesale/sum_within_month.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+def sum_within_month(
+    df: DataFrame, period_start_datetime: datetime, charge_type: ChargeType
+) -> DataFrame:
+    if charge_type.value == ChargeType.TARIFF.value:
+        qualities_expr = f.flatten(f.collect_set(Colname.qualities)).alias(
+            Colname.qualities
+        )
+    else:
+        qualities_expr = f.lit(None).alias(Colname.qualities)
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/calculation/wholesale/sum_within_month.py:33]: Maybe we should use f.lit(None).cast(ArrayType(StringType())).alias(Colname.qualities)? Logic tests might fail if you don't - I think.

File: source/databricks/calculation_engine/package/calculation/wholesale/sum_within_month.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            Colname.grid_area,
+            Colname.charge_key,
+            Colname.charge_code,
+            Colname.charge_type,
+            Colname.charge_owner,
+        )
+        .agg(
+            f.sum(Colname.total_amount).alias(Colname.total_amount),
+            f.sum(Colname.total_quantity).alias(Colname.total_quantity),
+            f.sum(Colname.charge_price).alias(Colname.charge_price),
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/calculation/wholesale/sum_within_month.py:47]: should we set this to None?

File: source/databricks/calculation_engine/tests/calculation/wholesale/test_subscription_calculators.py [hunk 3]
```diff
@@ -222,6 +225,38 @@ def test__returns_result_per_settlement_method(
         ]
         assert set(actual_settlement_methods) == set(expected)
 
+    def test__sum_within_month__sums_amount_per_month(
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/tests/calculation/wholesale/test_subscription_calculators.py:228]: what about a test where amount is None in one of the rows?

File: source/databricks/calculation_engine/package/calculation/wholesale/sum_within_month.py (hunk 2 above)
Comment (johevemi) [reply source/databricks/calculation_engine/package/calculation/wholesale/sum_within_month.py:47]: it is set to none in the select, but i will remove it from the .agg

File: source/databricks/calculation_engine/tests/calculation/wholesale/test_subscription_calculators.py (hunk 3 above)
Comment (johevemi) [reply source/databricks/calculation_engine/tests/calculation/wholesale/test_subscription_calculators.py:228]: i might combine the sum_within_month tests from tariff with subscription in its own file

File: source/databricks/calculation_engine/tests/calculation/wholesale/test_sum_within_month.py [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+        ChargeType.TARIFF,
+    )
+
+    # Assert
+    assert actual.collect()[0][Colname.total_amount] is None
+    assert actual.collect()[0][Colname.charge_price] is None
+    assert actual.count() == 1
+
+
+def test__sum_within_month__when_one_tariff_has_charge_price_none__sums_charge_price_and_total_amount_per_month_to_expected_value(
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/tests/calculation/wholesale/test_sum_within_month.py:309]: ????

File: source/databricks/calculation_engine/tests/calculation/wholesale/test_sum_within_month.py [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+
+
+def test__sum_within_month__sums_across_metering_point_types(
+    spark: SparkSession,
+) -> None:
+    # Arrange
+    rows = [
+        wholesale_results_factory.create_row(
+            charge_time=datetime(2020, 1, 1, 1),
+            charge_price=Decimal("2.000005"),
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/tests/calculation/wholesale/test_sum_within_month.py:124]: remove and use default?

File: source/databricks/calculation_engine/tests/calculation/wholesale/test_sum_within_month.py [hunk 6]
```diff
@@ -0,0 +1,404 @@
...
+        datetime(2019, 12, 31, 23),
+        ChargeType.TARIFF,
+    ).df
+
+    # Assert
+    assert actual.collect()[0][Colname.total_amount] == Decimal("4.020010")
+    assert actual.count() == 1
+
+
+def test__sum_within_month__tariff__joins_qualities(
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/tests/calculation/wholesale/test_sum_within_month.py:151]: Should always be none

**Review by energinet-ajw — APPROVED** [review]

**Pull request merged**